test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -696,4 +736,531 @@ private void failLoadWSGetWSInfoException(final Exception toThrow, final Excepti
             TestCommon.assertExceptionCorrect(got, expected);
         }
     }
+
+    @Test
+    public void updateEventAccessGroupEvent() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                "WS",
+                           Instant.ofEpochMilli(20000),
+                           StatusEventType.PUBLISH_ACCESS_GROUP)
+                           .withNullableAccessGroupID(1)
+                           .build();
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        // update event
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        // access group events are not updated, so we expect to get back the original event
+        Assert.assertEquals("expected the same event object", event, updatedEvent);
+    }
+
+    @Test
+    public void updateEventWorkspacePermanentlyDeleted() throws Exception {
```

Comment:
Seems like there should also be a test case for the `There is no workspace X`?

Reply:
to elaborate, my philosophy is that unit tests should 100% cover the code and get into all the nooks and crannies via mocks, and then integration tests make sure the various classes work together correctly, and so integration tests will necessarily be redundant on a per-class basis.

Reply:
I normally aim for 80-90% because of the law of diminishing returns but I can comply. This case was covered by the integration test but I have now added another test with mock